### PR TITLE
Update minimum NumPy version to v1.24.

### PR DESCRIPTION
**Pull request by hawkinsp**

### CODE REVIEW

**Comment by jakevdp** [comment]
Will this make JAX incompatible with tensorflow on Python 3.11 or older? https://github.com/tensorflow/tensorflow/blob/a76a162af4c24f11f035e7cf9ed39d45d51ed5bd/ci/official/requirements_updater/requirements.in#L2

**Comment by hawkinsp** [comment]
> Will this make JAX incompatible with tensorflow on Python 3.11 or older? https://github.com/tensorflow/tensorflow/blob/a76a162af4c24f11f035e7cf9ed39d45d51ed5bd/ci/official/requirements_updater/requirements.in#L2

I don't think so. I think those are the build versions.

If I download a `tensorflow` wheel and look at the METADATA, I see:
```
Requires-Dist: numpy <2.0.0,>=1.23.5 ; python_version <= "3.11"
Requires-Dist: numpy <2.0.0,>=1.26.0 ; python_version >= "3.12"
```

**Comment by jakevdp** [comment]
Ah, I see, this is the relevant requirements listing: https://github.com/tensorflow/tensorflow/blob/a76a162af4c24f11f035e7cf9ed39d45d51ed5bd/tensorflow/tools/pip_package/setup.py#L88

**Review by jakevdp — APPROVED** [review]

File: tests/lax_numpy_reducers_test.py [hunk 1]
```diff
@@ -507,7 +507,7 @@ def testReductionWithRepeatedAxisError(self):
       for weights_shape in ([None, shape] if axis is None or len(shape) == 1 or isinstance(axis, tuple)
                             else [None, (shape[axis],), shape])
     ],
-    keepdims=([False, True] if numpy_version >= (1, 23) else [None]),
+    keepdims=([False, True]),
```
Comment (jakevdp) [inline tests/lax_numpy_reducers_test.py:510]: We can remove handling of `keepdims=None` below.

**Pull request merged**